homeassistant/components/motion_blinds/gateway.py

```diff
@@ -48,3 +55,78 @@ async def async_connect_gateway(self, host, key):
             self.gateway_device.protocol,
         )
         return True
+
+    def check_interface(self):
+        """Check if the current interface supports multicast."""
+        with contextlib.suppress(socket.timeout):
+            return self.gateway_device.Check_gateway_multicast()
+        return False
+
+    async def async_get_interfaces(self):
+        """Get list of interface to use."""
+        interfaces = [DEFAULT_INTERFACE, "0.0.0.0"]
+        enabled_interfaces = []
+        default_interface = DEFAULT_INTERFACE
+
+        adapters = await network.async_get_adapters(self._hass)
+        for adapter in adapters:
+            if ipv4s := adapter["ipv4"]:
+                ip4 = ipv4s[0]["address"]
+                interfaces.append(ip4)
+                if adapter["enabled"]:
+                    enabled_interfaces.append(ip4)
+                    if adapter["default"]:
+                        default_interface = ip4
+
+        if len(enabled_interfaces) == 1:
+            default_interface = enabled_interfaces[0]
+            interfaces.remove(default_interface)
+            interfaces.insert(0, default_interface)
+
+        if self._interface is not None:
+            interfaces.remove(self._interface)
+            interfaces.insert(0, self._interface)
+
+        return interfaces
+
+    async def async_check_interface(self, host, key):
+        """Connect to the Motion Gateway."""
+        self._interfaces = await self.async_get_interfaces()
```

Comment:
It looks like this is stored but never accessed again outside of this function?

Reply:
you are right, I changed it to a local variable